web_src/js/modules/tippy.js

```diff
@@ -2,12 +2,7 @@ import tippy from 'tippy.js';
 
 const visibleInstances = new Set();
 
-export function createTippy(target, opts = {}) {
-  const {role, content, onHide: optsOnHide, onDestroy: optsOnDestroy, onShow: optOnShow} = opts;
-  delete opts.onHide;
-  delete opts.onDestroy;
-  delete opts.onShow;
-
+export function createTippy(target, {onHide, onShow, onDestroy, role, content, ...other} = {}) {
```

Comment:
Well, this is wrong now (that's what I said why it is really unclear).

Because the `role` is exacted, `other` doesn't contain `role`. Then how do you use `...other` to override the default `role` ....

Reply:
The only change that was needed would have been `role: role || 'menu'`. I still want to do my method.

Reply:
I think current style is the best, the reasons are:

1. `export function createTippy(target, opts = {})` is shorter and clearer for readers, it's easy to catch the function declaration.
2. I remember that you mentioned that you don't want to read a line with more than 100 chars, using destructing in the function declaration makes the line longer and longer.
3. If you destructure "role", this time you can patch the code by `role: role || 'menu'`, next time if another variable is destructed, developers would forget the `...other` overriding again. It is quite fragile and unclear. For example, I have mentioned that `the code would be more complex for using "role" later.`, the changed code still contains bugs, so it is a complex and counterintuitive problem.

Reply:
Well I guess we can keep it.